### DISCUSSION TITLE
建议：对更多格式的ip进行支持

### DISCUSSION THREAD

**Discussion by fengguowudi** [XIU2/CloudflareSpeedTest#60] [讨论]
例：
一：162.159.208.4-162.159.208.103
二：172.64.32.*

**Comment by XIU2** [comment]
意义不大，这样太麻烦了。

不能随便创造一种格式，我都要去支持，所以还是按 **IP 段的标准格式**走吧。

****

即使是对 IP 段标准格式（子网掩码）不熟，网上也有**在线 IP 段计算器**，可以轻松计算出来的。

**Comment by fengguowudi** [comment]
但是我不会转换诶⊙﹏⊙b汗
您可以教教我怎么转换吗
172.64.32.*
这种的意思就是这个*是0-255
这个
162.159.208.4-162.159.208.103
则是末尾4-103
请问该怎么转换？

**Reply by XIU2** [reply]
```
172.64.32.0/24
162.159.208.0/25
# 你说的 4~103 是无法用 IP 段子网掩码去表达的，不符合规范，所以只能用 /25 代替了，也就是 1~126
```

**Comment by fengguowudi** [comment]
172.64.32.0/24这种写法我单独放入ip.txt里显示只有4个ip并且没有测速直接退出，请问是为什么呢？

**Reply by XIU2** [reply]
软件为了测速速度（效率），会对每个 IP 段只随机取一个 IP 进行测速的（指的是 IP 最后一段的 .XXX ）。
你看到的 4 指的是 IP 数（1 个） × 延迟测速次数（默认 4 次）。

如果你想要该 IP 段的每个 IP 都进行测速，请在运行时加上 `-allip` 参数。

****

直接退出是因为该 IP 延迟测速结果为超时，所以软件就直接退出了。

**Reply by XIU2** [reply]
如果用着还不错，就给项目点个 ⭐ 吧！差一点就一千个 ⭐ 了...
